DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs

```diff
@@ -390,26 +392,44 @@ public IActionResult Tutorial(int customisationId, int sectionId, int tutorialId
                 logger.LogError(
                     "Redirecting to 404 as customisation/section/tutorial id was not found. " +
                     $"Candidate id: {candidateId}, customisation id: {customisationId}, " +
-                    $"centre id: {centreId.ToString() ?? "null"}, section id: {sectionId} tutorial id: {tutorialId}");
+                    $"centre id: {centreId.ToString() ?? "null"}, section id: {sectionId} tutorial id: {tutorialId}"
+                );
                 return RedirectToAction("StatusCode", "LearningSolutions", new { code = 404 });
             }
+
             if (!String.IsNullOrEmpty(tutorialInformation.Password) && !tutorialInformation.PasswordSubmitted)
             {
                 return RedirectToAction("CoursePassword", "LearningMenu", new { customisationId });
             }
+
             var progressId = courseContentService.GetOrCreateProgressId(candidateId, customisationId, centreId);
 
             if (progressId == null)
             {
                 logger.LogError(
                     "Redirecting to 404 as no progress id was returned. " +
-                    $"Candidate id: {candidateId}, customisation id: {customisationId}, centre id: {centreId}");
+                    $"Candidate id: {candidateId}, customisation id: {customisationId}, centre id: {centreId}"
+                );
                 return RedirectToAction("StatusCode", "LearningSolutions", new { code = 404 });
             }
 
             sessionService.StartOrUpdateDelegateSession(candidateId, customisationId, HttpContext.Session);
             courseContentService.UpdateProgress(progressId.Value);
 
+            /* Course progress doesn't get updated if the auth token expires by the end of the tutorials. 
+              Some tutorials are longer than the default auth token lifetime, so we set the auth expiry to 8 hours.
+              See HEEDLS-637 and HEEDLS-674 for more details */
+            if (tutorialInformation.AverageTutorialDuration >= 45)
+            {
+                var authProperties = new AuthenticationProperties
+                {
+                    AllowRefresh = true,
+                    IssuedUtc = DateTime.UtcNow,
+                    ExpiresUtc = DateTime.UtcNow.AddHours(8)

```

Comment:
Yes, our thinking was that if a tutorial is over a certain length, then some people will take breaks / intersperse it with other work. So it would be good to let them come back to it anytime over their working day.

Agree with the clockService comment: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/20067450883/Dealing+with+time